cmake/HPX_AddConfigTest.cmake

```diff
@@ -57,14 +57,14 @@ macro(add_hpx_config_test variable)
             "-DCOMPILE_DEFINITIONS=${CONFIG_TEST_COMPILE_DEFINITIONS}"
           RUN_OUTPUT_VARIABLE ${variable}_OUTPUT
           ARGS ${${variable}_ARGS})
-          if(${variable}_COMPILE_RESULT AND NOT ${variable}_RUN_RESULT)
-            set(${variable}_RESULT TRUE)
-          else()
-            set(${variable}_RESULT FALSE)
```

Comment:
What happens if the result hasn't been run? is `${variable}_RUN_RESULT` set to `FALSE` or `TRUE`?
Are all config tests automatically set to `FALSE` if we are cross compiling now?